java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java

```diff
@@ -55,35 +54,60 @@ private RuleEntryParams getRuleEntryParams(String commandId, List<Object> argume
         return params;
     }
 
-    private static List<SymbolInformation> search(String projectName, String query, int location, IProgressMonitor monitor) {
-        IJavaProject[] targetProjects;
-        IJavaProject project = ProjectUtils.getJavaProject(projectName);
-        logInfo("Searching in project: " + project + " Query: " + query);
-        if (project != null) {
-			targetProjects = new IJavaProject[] { project };
-		} else {
-			targetProjects= ProjectUtils.getJavaProjects();
-		}
+    // mapLocationToSearchPatternLocation will create the correct search pattern or throw an error if one can not be built.
+    // the search patterns are responsible for finding as many locations/symbols as possible. We will relay on the client
+    // to filter.
+    private static SearchPattern mapLocationToSearchPatternLocation(int location, String query) throws Exception {
+        //TODO: Normalize queries and/or verify for each location.
+        switch (location) {
+        // Using type for both type and annotation.
+        case 0:
+        case 4:
+            return SearchPattern.createPattern(query, IJavaSearchConstants.TYPE, IJavaSearchConstants.ALL_OCCURRENCES, SearchPattern.R_PATTERN_MATCH);
+        case 1:
+            throw new Exception("Inheritance is not implemented");
+        case 2: 
+            return SearchPattern.createPattern(query, IJavaSearchConstants.METHOD, IJavaSearchConstants.ALL_OCCURRENCES, SearchPattern.R_PATTERN_MATCH);
+        case 3:
+            return SearchPattern.createPattern(query, IJavaSearchConstants.CONSTRUCTOR, IJavaSearchConstants.ALL_OCCURRENCES, SearchPattern.R_PATTERN_MATCH);
+        }
+        throw new Exception("unable to create search pattern"); 
+    }
 
-		int s = IJavaSearchScope.REFERENCED_PROJECTS | IJavaSearchScope.SOURCES | IJavaSearchScope.APPLICATION_LIBRARIES | IJavaSearchScope.SYSTEM_LIBRARIES;
+    private static List<SymbolInformation> search(String projectName, String query, int location, IProgressMonitor monitor) throws Exception {
+        JavaModelManager.getIndexManager().waitForIndex(true, monitor);
+        DeltaProcessingState state = JavaModelManager.getDeltaState();
+        logInfo("sourceAttachements: " + state.sourceAttachments);
+
+        Map<IPath, IPath> attachments = state.sourceAttachments;
+        Thread.sleep(5000);
+        while (attachments.size() != state.sourceAttachments.size()) {
+            logInfo("waiting size: " + state.sourceAttachments.size());
+            attachments = state.sourceAttachments;
+            Thread.sleep(10000);
```

Comment:
Just wondering if we know the ideal alternative solution for sleep()?

Reply:
@pranavgaikwad no, I think it's going to be kind of tough unfortunately. We may want to toss some TODOs on those though so we don't forget.

Reply:
added todo and issue